File: CHUONG_5/5.15_ViewPager_Exercise/app/src/main/java/com/example/viewpager/data/repository/ZodiacRepositoryImpl.java
package com.example.viewpager.data.repository;

import com.example.viewpager.data.datasource.ZodiacLocalDataSource;
import com.example.viewpager.data.model.Zodiac;

import java.util.List;

public class ZodiacRepositoryImpl implements ZodiacRepository{
    private final ZodiacLocalDataSource mLocalDataSource;

    public ZodiacRepositoryImpl(ZodiacLocalDataSource dataSource) {
        mLocalDataSource = dataSource;
    }
    @Override
    public List<Zodiac> getZodiacs() {

        return mLocalDataSource.getZodiacs();
    }
}

File: CHUONG_5/5.15_Zodiac_kotlin/app/src/main/java/com/example/zodiac/data/repository/ZodiacRepositoryImpl.kt
package com.example.zodiac.data.repository

import com.example.zodiac.data.datasource.ZodiacLocalDataSource
import com.example.zodiac.data.model.Zodiac

class ZodiacRepositoryImpl(
    private val localDataSource: ZodiacLocalDataSource
): ZodiacRepository {
    override fun getZodiacs(): List<Zodiac> {
        return localDataSource.getZodiacs()
    }
}